lib/widgets/lightbox.dart

```diff
@@ -208,11 +213,244 @@ class _LightboxPageState extends State<_LightboxPage> {
+          text: TextSpan(children: [
+            TextSpan(
+              text: '${widget.message.senderFullName}\n',
+
+              // Restate default
+              style: themeData.textTheme.titleLarge!.copyWith(color: appBarForegroundColor)),
+            TextSpan(
+              text: timestampText,
+
+              // Make smaller, like a subtitle
+              style: themeData.textTheme.titleSmall!.copyWith(color: appBarForegroundColor)),
+          ])));
+    }
+
+    Widget? bottomAppBar;
+    if (_controller != null && _headerFooterVisible) {
+      bottomAppBar = BottomAppBar(
+        height: 150,
+        color: appBarBackgroundColor,
+        elevation: appBarElevation,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+          Row(
+            children: [
+              Text(
+                _controller!.value.position.formatHHMMSS(),
+                style: const TextStyle(color: Colors.white),
+              ),
+              Expanded(
+                child: Slider(
+                  value: _controller!.value.position.inSeconds.toDouble(),
+                  max: _controller!.value.duration.inSeconds.toDouble(),
+                  activeColor: Colors.white,
+                  onChanged: (value) {
+                    _controller!.seekTo(Duration(seconds: value.toInt()));
+                  },
+                ),
+              ),
+              Text(
+                 _controller!.value.duration.formatHHMMSS(),
+                style: const TextStyle(color: Colors.white),
+              ),
+            ],
+          ),
+          IconButton(
+            onPressed: () {
+              if (_controller!.value.isPlaying) {
+                _controller!.pause();
+              } else {
+                _controller!.play();
+              }
+            },
+            icon: Icon(
+              _controller!.value.isPlaying
+                ? Icons.pause_circle_rounded
+                : Icons.play_circle_rounded,
+              size: 50,
+            )),
+        ]));
+    }
+
+    return Theme(
+      data: themeData.copyWith(
+        iconTheme: themeData.iconTheme.copyWith(color: appBarForegroundColor)),
+      child: Scaffold(
+        backgroundColor: Colors.black,
+        extendBody: true, // For the BottomAppBar
+        extendBodyBehindAppBar: true, // For the AppBar
+        appBar: appBar,
+        bottomNavigationBar: bottomAppBar,
+        body: MediaQuery(
+          // Clobber the MediaQueryData prepared by Scaffold with one that's not
+          // affected by the app bars. On this screen, the app bars are
+          // translucent, dismissible overlays above the pan-zoom layer in the
+          // Z direction, so the pan-zoom layer doesn't need avoid them in the Y
+          // direction.
+          data: MediaQuery.of(context),
+
+          child: GestureDetector(
+            behavior: HitTestBehavior.translucent,
+            onTap: _handleTap,
```

Comment:
There's a lot of structure here — basically everything from `return` through this `onTap` line, and some other code that supports that — that's duplicated between the two kinds of lightbox. I think that's not a coincidence, in that it's basically the code for making "a lightbox" independent of exactly what's shown in the lightbox.

So let's factor things out to deduplicate that code. Two reasonable strategies are:
* Make a common base class, or I guess pair of base classes, like `abstract class _LightboxPage extends StatefulWidget` and `abstract class _LightboxPageState extends State<_LightboxPage>`. Then the subclasses don't override `build`, but rather some subroutines that `build` calls, like `buildBottomAppBar` and `buildChild`. (Plus `initState` etc.)
  This is the approach taken at `MessageActionSheetMenuItemButton` and some other places.
* Make a common helper widget, like `class _LightboxPageLayout extends StatefulWidget`. Then the specific lightbox widgets pass constructor arguments like `buildBottomAppBar` and `child`. The state lifetime methods like `initState` etc. are independent on each widget and don't have to interact via `super` calls.

  This is the approach taken at `_ComposeBoxLayout` and I think some other places.